src/main/java/org/minions/devfund/carlosorellana/WhackAMole.java

```diff
@@ -0,0 +1,170 @@
+        this.moleGrid = new char[gridDimension][gridDimension];
+
+        for (char[] mole : this.moleGrid) {
+            Arrays.fill(mole, NO_MOLE_CHAR);
+        }
+    }
+
+    /**
+     * Sends the Mole positions in the Grid.
+     *
+     * @param x - X positions.
+     * @param y - Y positions.
+     * @return True if the position is correct, otherwise false.
+     */
+    public boolean place(final int x, final int y) {
+        int xPosition = x - 1;
+        int yPosition = y - 1;
+
+        if (isCorrectPosition(xPosition, yPosition)
+             && moleGrid[xPosition][yPosition] != MOLE_CHAR) {
+                moleGrid[xPosition][yPosition] = MOLE_CHAR;
+                molesLeft++;
+                return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Validate if the coordinate is correct in the grid.
+     *
+     * @param x - X position.
+     * @param y - Y position.
+     * @return True fi the position is correct in the grid, otherwise false.
+     */
+    private boolean isCorrectPosition(final int x, final int y) {
+        return x >= 0 && x < moleGrid.length && y >= 0 && y < moleGrid.length;
+    }
+
+    /**
+     * Sends the Whack position in the Grid.
+     *
+     * @param x - X position.
+     * @param y - Y positions.
+     */
+    public void whack(final int x, final int y) {
+        int xPosition = x - 1;
+        int yPosition = y - 1;
+
+        if (attemptsLeft <= 0) {
+            messageAttempts = "The number of attempts exceeds the allowed";
+            return;
+        }
+
+        if (attemptsLeft > 0) {
+            if (isCorrectPosition(xPosition, yPosition) && moleGrid[xPosition][yPosition] == MOLE_CHAR) {
+                moleGrid[xPosition][yPosition] = WHACKED_CHAR;
+                score++;
+                attemptsLeft--;
+                molesLeft--;
+            } else {
+                attemptsLeft--;
+            }
+        }
+    }
+
+    /**
+     * Prints the Grid to User.
+     *
+     * @return String value.
+     */
+    public String printGridToUser() {
+        StringBuilder gridToUser = new StringBuilder();
+
+        for (char[] row : moleGrid) {
+            for (char place : row) {
+                char currentPlace = place == WHACKED_CHAR ? WHACKED_CHAR : NO_MOLE_CHAR;
+                gridToUser.append(String.valueOf(currentPlace)).append(SPACE);
+            }
+            gridToUser.append(NEW_LINE);
+        }
+        System.out.println(gridToUser.toString());
+        return gridToUser.toString();
+    }
+
+    /**
+     * Prints the Grid initial.
+     *
+     * @return String value.
+     */
+    public String printGrid() {
+        StringBuilder gridBuilder = new StringBuilder();
+
+        for (char[] row : moleGrid) {
+            for (char value : row) {
+                gridBuilder.append(String.valueOf(value)).append(SPACE);
+            }
+            gridBuilder.append(NEW_LINE);
+        }
+        System.out.println(gridBuilder.toString());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)